### DISCUSSION TITLE
Went through the docu and forum, but still have questions

### DISCUSSION THREAD

**Discussion by Stitch10925** [seaweedfs/seaweedfs#2826] [Q&A] [answered]
Hey all,

I have been very interested in distributed network filesystems and had my eyes set on Ceph for a long time since Proxmox, the hypervisor I am using, has support for it built in. However, it seems too complex for my Linux-noob mind to wrap my head around. I hit upon SeaweedFS and it seems nearly perfect for what I need, plus it's not too complex. I went through the documentation but still have some questions about the setup:

1. I understand the concept of a master which speaks to volume servers to write and read files. I also understand that a Filer is an added layer on top of the master that allows you to connect to SeaweedFS in different ways, however, it seems like you can talk to the master or the volume directly using the API. If so, does the Volume sync back to the master to tell it what has changed? Wouldn't it be better to just write to the master?
2. I am failing to see how SeaweedFS is setup on hardware level: So, I have a dedicated master and some storage servers. On these storage servers, do I create a volume per server? Do create a volume per hard drive? Do I have to create a volume per 30 GB?
3. Would it make sense, in case of 3 storage servers, so install a master on them together with the volumes for redundancy purposes or will that hurt performance too much?
4. What are the hardware requirements to run SeaweedFS? Can I run it on a gigabit network or do I need higher speeds? What hardware is required for the master or storage servers?
5. Does SeaweedFS read files from multiple volumes at the same time to increase throughput?

Sorry for all the question. I'm trying to get a good grasp of all the concepts before I try and start moving my data around.

**Comment by chrislusf** [comment, marked as answer]
1. Not.
2. "volume" in SeaweedFS is just a normal file. You can think it is a virtual hard drive. You can create many "SeaweedFS volumes" in one hard drive.
3. You can add many more storage servers. The system is designed to read/write to the storage servers in parallel for better performance.
4. No hardware requirements. All depends on business requirements, whether it is for home usage, or big data center, or low latency, or capacity, etc.
5. Yes.

**Reply by Stitch10925** [reply]
Hi Chris,

Thank you for the reply.

Not sure what you meant by "Not" in point 1. Not as in: The volume does not sync back to the master? Or not as in: It would not be better to only write to the master?

As for #2: How do you determine how many volumes to run? I assume if replication is enabled you would not want to have multiple volumes on the same hard drive since then the hard drive would still be a single point of failure. Is it better it better to have many small volumes or few big ones? Or would it be best to create a volume per hard drive?

Thanks

**Reply by chrislusf** [reply]
1. the Volume does not sync back to the master to tell it what has changed.
2. If the files are not updated/deleted, you can create larger volumes. If not, the volumes need to be compacted periodically, so better not to keep it too large. You can create a few volumes, around 20~50, for example. Also, each bucket needs a few of volumes. Need to prepare for that also.

**Reply by Stitch10925** [reply]
So for cold(-ish) storage like movies, music, pictures it is fine to create larger volumes, but if it is small, changing data like word documents, source control, etc. it makes more sense to create multiple smaller volumes?

I am not planning on using cloud-storage, so I probably won't have to worry about the buckets then? Not quite sure I understand the concept of buckets.

**Reply by chrislusf** [reply]
correct.
